pkg/storegateway/gateway_blocks_http.go

```diff
@@ -109,24 +110,38 @@ func (s *StoreGateway) BlocksHandler(w http.ResponseWriter, req *http.Request) {
 			blockSplitID = &bsc
 		}
 		lbls := labels.FromMap(m.Thanos.Labels)
+		noCompactDetails := []string{}
+		if val, ok := noCompactMarkerDetails[m.ULID]; ok {
+			noCompactDetails = []string{
+				fmt.Sprintf("Time: %s", formatTimeIfNotZero(time.Unix(val.NoCompactTime, 0), time.RFC3339)),
+				fmt.Sprintf("Reason: %s", val.Reason),
+			}
+		}
+
+		deletedTime := ""
+		if dt, ok := deleteMarkerDetails[m.ULID]; ok {
+			deletedTime = formatTimeIfNotZero(time.Unix(dt.DeletionTime, 0).UTC(), time.RFC3339)
+		}
```

Comment:
the `if` shouldn't be necessary if `formatTimeIfNotZero` already checks for zero values

Reply:
it was not really the case, when i don't check it, i do see 1970-01-01T00:00:00Z on UI, I think here is the default 0 convert to UTC with the time format doesn't really give a zero value. A bit confusing, I am still checking why

Reply:
it is not the same, as discussed, changed formatTimeIfNotZero function instead